src/bindings/js/node/src/model_wrap.cpp

```diff
@@ -154,3 +156,20 @@ Napi::Value ModelWrap::get_friendly_name(const Napi::CallbackInfo& info) {
     const auto friendly_name = _model->get_friendly_name();
     return Napi::String::New(env, friendly_name);
 }
+
+Napi::Value ModelWrap::get_output_shape(const Napi::CallbackInfo& info) {
+    if (info.Length() != 1 || !info[0].IsNumber()) {
+        reportError(info.Env(), "Invalid argument. Expected a single number.");
+        return info.Env().Undefined();
+    }
+
+    auto idx = info[0].As<Napi::Number>().Int32Value();
+    auto cm_outputs = _model->outputs();  // Output<Node>
+
+    try {
+        return cpp_to_js<ov::Shape, Napi::Array>(info, cm_outputs[idx].get_shape());
```

Comment:
If you take `cm_outputs[idx].get_shape()` where idx will be out of range it will lead to exception that won't be catch (for some reason). 
Because of that I propose to check `cm_output` size first.

Reply:
Even better to use `_model->output(idx)` instead of getting all outputs and select one of them.